dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,851 @@
+            .getSystemUser(), false, limit, offset);
+        int size = contentlets.size();
+        while (size > 0) {
+            for (Contentlet contentlet : contentlets) {
+                invalidate(contentlet);
+            }
+            offset += limit;
+            contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+                .getSystemUser(), false, limit, offset);
+            size = contentlets.size();
+        }
+    }
+
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, PageMode mode, String language, String filePath) {
+        try {
+
+            Identifier identifier = APILocator.getIdentifierAPI()
+                .find(id1);
+            Contentlet contentlet = null;
+            if (CacheLocator.getVeloctyResourceCache()
+                .isMiss(filePath)) {
+                if (LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+                    LanguageAPI langAPI = APILocator.getLanguageAPI();
+                    language = Long.toString(langAPI.getDefaultLanguage()
+                        .getId());
+                } else {
+                    throw new ResourceNotFoundException("Contentlet is a miss in the cache");
+                }
+            }
+
+            try {
+                contentlet = APILocator.getContentletAPI()
+                    .findContentletByIdentifier(identifier.getId(), mode.showLive, new Long(language), APILocator.getUserAPI()
+                        .getSystemUser(), true);
+            } catch (DotContentletStateException e) {
+                contentlet = null;
+            }
+
+            if (contentlet == null || !InodeUtils.isSet(contentlet.getInode()) || contentlet.isArchived()) {
+
+                LanguageAPI langAPI = APILocator.getLanguageAPI();
+                long lid = langAPI.getDefaultLanguage()
+                    .getId();
+                if (lid != Long.parseLong(language)) {
+                    Contentlet cc = APILocator.getContentletAPI()
+                        .findContentletByIdentifier(identifier.getId(), mode.showLive, lid, APILocator.getUserAPI()
+                            .getSystemUser(), true);
+                    if (cc != null && UtilMethods.isSet(cc.getInode()) && !cc.isArchived()
+                            && LanguageWebAPI.canApplyToAllLanguages(cc)) {
+                        contentlet = cc;
+                    } else {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(filePath);
+                        throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+                    }
+                }
+            }
+
+            Logger.debug(this, "DotResourceLoader:\tWriting out contentlet inode = " + contentlet.getInode());
+            return buildVelocity(contentlet, identifier, mode, filePath);
+
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)